### PR TITLE
feat(v2): hide navbar on scroll

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

I propose to consider including in my opinion a good feature called "Auto-Hide Sticky Header". The bottom line is simple - navbar hides when scrolling down, and is shown again when scrolling up. This helps the user to better _focus_ on the content. Since not everyone may like it, this option is enabled in the theme config.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

The `hideOnScroll` option must be enabled.

```js
  themeConfig: {
    navbar: {
      hideOnScroll: true,
```

AwesomeScreenshot-2019-11-25-1574707306503

### CODE REVIEW

**Comment by lex111** [comment]
Hmm, for several days there has been no review from anyone, I think this is a good addition, don't you like it?

File: packages/docusaurus-theme-classic/src/theme/Navbar/styles.module.css [hunk 1]
```diff
@@ -16,3 +16,11 @@
     display: none;
   }
 }
+
+.navbarHidable {
+  transition: top 0.2s ease-in-out;
+}
+
+.navbarHided {
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/Navbar/styles.module.css]: Nit grammar - `navbarHidden`

File: packages/docusaurus-theme-classic/src/theme/Navbar/styles.module.css [hunk 2]
```diff
@@ -16,3 +16,11 @@
     display: none;
   }
 }
+
+.navbarHidable {
```
Comment (yangshun) [inline packages/docusaurus-theme-classic/src/theme/Navbar/styles.module.css]: Nit spelling - `navbarHideable`

**Comment by endiliey** [comment]
Docs needed

**Comment by lex111** [comment]
Sorry for the delay, all proposals implemented (hook, docs).

File: website/docs/theme-classic.md [hunk 3]
```diff
@@ -90,6 +90,23 @@ module.exports = {
 
 Outbound links automatically get `target="_blank" rel="noopener noreferrer"`.
 
+### Auto-hide sticky navbar
+
+You can enable this cool UI feature that automatically hides the navbar when a user starts scrolling down the page, and show it again when the user scrolls up.
```
Comment (wgao19) [inline website/docs/theme-classic.md]: it feels a bit weird that here we first address reader by "you" then later on say "a user" or "the user". I think it is ok to keep using you.

just personal opinion: i prefer it to be less objective. "cool" is also a very vague adjective and i found it a bit weird to read this from our docs

File: packages/docusaurus-theme-classic/src/theme/hooks/useHideableNavbar.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState, useCallback, useEffect} from 'react'; // eslint-disable-line no-unused-vars
```
Comment (endiliey) [inline packages/docusaurus-theme-classic/src/theme/hooks/useHideableNavbar.js]: why need eslint disable ? is it React :O

**Pull request merged**